internal/services/datadog/datadog_monitors_resource.go

```diff
@@ -0,0 +1,493 @@
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogUsersName,
+						},
+
+						"email_address": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogMonitorsEmailAddress,
+						},
+
+						"phone_number": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogMonitorsPhoneNumber,
+						},
+					},
+				},
+			},
+
+			"monitoring_status": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"liftr_resource_category": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"liftr_resource_preference": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"marketplace_subscription_status": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+func resourceDatadogMonitorCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Datadog.MonitorsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	id := parse.NewDatadogMonitorID(subscriptionId, resourceGroup, name).ID()
+
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing Datadog Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if !utils.ResponseWasNotFound(existing.Response) {
+		return tf.ImportAsExistsError("azurerm_datadog_monitor", id)
+	}
+
+	monitoringStatus := datadog.MonitoringStatusDisabled
+	if d.Get("monitoring_status").(bool) {
+		monitoringStatus = datadog.MonitoringStatusEnabled
+	}
+
+	body := datadog.MonitorResource{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Identity: expandMonitorIdentityProperties(d.Get("identity").([]interface{})),
+		Sku:      expandMonitorResourceSku(d.Get("sku").([]interface{})),
+		Properties: &datadog.MonitorProperties{
+			DatadogOrganizationProperties: expandMonitorOrganizationProperties(d.Get("datadog_organization_properties").([]interface{})),
+			UserInfo:                      expandMonitorUserInfo(d.Get("user_info").([]interface{})),
+			MonitoringStatus:              monitoringStatus,
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+	future, err := client.Create(ctx, resourceGroup, name, &body)
+	if err != nil {
+		return fmt.Errorf("creating Datadog Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
We can simplify this to return fmt.Errorf("creating %s: %+v", id, err), because the necessary information are included in id.

Reply:
Incorporating in next commit.